apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx

```diff
@@ -112,33 +156,79 @@ export default function ResponseOptionsCard({ localSurvey, setLocalSurvey }: Res
-                  <div className="ml-2">
-                    <h3 className="text-sm font-semibold text-slate-700">Redirect on completion</h3>
+          <div className="p-3 ">
+            <div className="ml-2 flex items-center space-x-1">
+              <Switch id="redirectUrl" checked={redirectToggle} onCheckedChange={handleRedirectCheckMark} />
+              <Label htmlFor="redirectUrl" className="cursor-pointer">
+                <div className="ml-2">
+                  <h3 className="text-sm font-semibold text-slate-700">Redirect on completion</h3>
+                  <p className="text-xs font-normal text-slate-500">
+                    Redirect user to specified link on survey completion
+                  </p>
+                </div>
+              </Label>
+            </div>
+            {redirectToggle && (
+              <div className="mt-4">
+                <Input
+                  type="url"
+                  placeholder="https://www.example.com"
+                  value={redirectUrl ? redirectUrl : ""}
+                  onChange={(e) => handleRedirectUrlChange(e.target.value)}
+                />
+              </div>
+            )}
+          </div>
+          <div className="p-3 ">
+            <div className="ml-2 flex items-center space-x-1">
+              <Switch
+                id="surveyDeadline"
+                checked={surveyCloseOnDateToggle}
+                onCheckedChange={handleSurveyCloseOnDateToggle}
+              />
+              <Label htmlFor="surveyDeadline" className="cursor-pointer">
+                <div className="ml-2">
+                  <h3 className="text-sm font-semibold text-slate-700">Close survey on date</h3>
+                  {localSurvey.status === "completed" && (
                     <p className="text-xs font-normal text-slate-500">
-                      Redirect user to specified link on survey completion
+                      This form is already completed. You can change the status settings to make best use of
+                      this option.
                     </p>
-                  </div>
-                </Label>
-              </div>
+                  )}
+                </div>
+              </Label>
+            </div>
+            {surveyCloseOnDateToggle && (
               <div className="mt-4">
-                {redirectToggle && (
-                  <Input
-                    type="url"
-                    placeholder="https://www.example.com"
-                    value={redirectUrl ? redirectUrl : ""}
-                    onChange={(e) => handleRedirectUrlChange(e.target.value)}
-                  />
-                )}
+                <DatePicker date={closeOnDate} handleDateChange={handleCloseOnDateChange} />
               </div>
-            </div>
-          )}
+            )}
+          </div>
         </div>
       </Collapsible.CollapsibleContent>
     </Collapsible.Root>
   );
 }
+
+export function DatePicker({ date, handleDateChange }) {
```

Comment:
Please move the datepicker to a separate, reusable component :)

Reply:
sure, will do this.

Reply:
thanks!